k8s-deploy/pkg/job/cluster_install.go

```diff
@@ -0,0 +1,363 @@
+	}
+	job = &j
+	if job.Status != modules.JobStatusPending {
+		return errors.New("job.Status error")
+	}
+	dbErr = job.SetStatus(modules.JobStatusRunning)
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job setStatus error")
+		return dbErr
+	}
+	dbErr = job.SetState("job start Running")
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job setResult error")
+		return dbErr
+	}
+	return nil
+}
+
+func helmInstall(job *modules.Job, cluster *modules.Cluster) error {
+
+	clusterCover(job, cluster)
+
+	err := cluster.HelmInstall()
+	if err != nil {
+		log.Error().Err(err).Str("ClusterId", cluster.Uuid).Msg("helm install cluster error")
+
+		dbErr := job.SetStatus(modules.JobStatusFailed)
+		if dbErr != nil {
+			log.Error().Err(dbErr).Msg("job.SetStatus error")
+		}
+		return err
+	}
+
+	dbErr := job.SetStatus(modules.JobStatusRunning)
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job.SetStatus error")
+	}
+	return nil
+
+}
+
+func clean(job *modules.Job, cluster *modules.Cluster) {
+
+	_, err := cluster.Delete()
+	if err != nil {
+		log.Error().Err(err).Interface("cluster", cluster).Msg("DB delete cluster error")
+	}
+	err = cluster.HelmDelete()
+	if err != nil {
+		log.Error().Err(err).Interface("cluster", cluster).Msg("helm delete cluster error")
+	}
+
+	log.Debug().Str("cluster uuid", cluster.Uuid).Msg("install cluster rollBACK success")
+}
+
+func addJobEvent(job *modules.Job, Event string) {
+
+	job.States = append(job.States, Event)
+
+	dbErr := job.SetStates(job.States)
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job.SetStatus error")
+	}
+	return
+}
+func updateLastJobEvent(job *modules.Job, Event string) {
+
+	job.States[len(job.States)-1] = Event
+
+	dbErr := job.SetStates(job.States)
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job.SetStatus error")
+	}
+	return
+}
+func checkStatus(job *modules.Job, cluster *modules.Cluster) bool {
+
+	// update subJobs
+	ClusterStatus, err := service.GetClusterPodStatus(cluster.Name, cluster.NameSpace)
+	if err != nil {
+		log.Error().Err(err).Msg("GetClusterPodStatus error")
+	}
+
+	log.Debug().Interface("ClusterStatus", ClusterStatus).Msg("GetClusterPodStatus()")
+	subJobs := generateSubJobs(job, ClusterStatus)
+
+	dbErr := job.SetSubJobs(subJobs)
+	if dbErr != nil {
+		log.Error().Err(dbErr).Msg("job setSubJobs error")
+	}
+
+	if service.CheckClusterStatus(ClusterStatus) {
+		dbErr := job.SetStatus(modules.JobStatusSuccess)
+		if dbErr != nil {
+			log.Error().Err(dbErr).Msg("job setStatus error")
+		}
+		return true
+	}
+	return false
+}
+
+func clusterCover(job *modules.Job, cluster *modules.Cluster) {
+	// Override existing installation
+	if job.Metadata.Cover {
+		log.Info().Msg("overwrite current installation")
+		err := cluster.HelmDelete()
+		if err != nil {
+			log.Error().Err(err).Msg("helm clean error")
+		}
+		addJobEvent(job, "overwrite current installation")
+		log.Info().Str("name", cluster.Name).Str("namespace", cluster.NameSpace).Msg("HelmClean success")
+	}
+}
```

Comment:
A general comments: there are quite some inconsistency in the terms styles like:

`[S|s]uccess`  
`[C|c]luster`  
`[B|b]y`
`[R|r]unning`
...

maybe we should check all these wording and make them consistent.